Kentico.Kontent.Management/Models/LanguageVariants/Elements/BaseElement.cs

```diff
@@ -21,34 +25,140 @@ public abstract class BaseElement
+            };
+        }
+        else if (type == typeof(DateTimeElement))
+        {
+            return new DateTimeElement
+            {
+                Element = Reference.FromDynamic(source.element),
+                Value = Convert.ToDateTime(source.value)
+            };
+        }
+        else if (type == typeof(LinkedItemsElement))
+        {
+            return new LinkedItemsElement
+            {
+                Element = Reference.FromDynamic(source.element),
+                Value = (source.value as IEnumerable<dynamic>)?.Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id)))
+            };
+        }
+        else if (type == typeof(MultipleChoiceElement))
+        {
+            return new MultipleChoiceElement
+            {
+                Element = Reference.FromDynamic(source.element),
+                Value = (source.value as IEnumerable<dynamic>)?.Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id)))
+            };
+        }
+        else if (type == typeof(TaxonomyElement))
+        {
+            return new TaxonomyElement
+            {
+                Element = Reference.FromDynamic(source.element),
+                Value = (source.value as IEnumerable<dynamic>)?.Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id)))
+            };
+        }
+        else if (type == typeof(UrlSlugElement))
+        {
+            return new UrlSlugElement
+            {
+                Element = Reference.FromDynamic(source.element),
+                Mode = source.mode?.ToString(),
+                Value = source.value?.ToString()
+            };
+        }
+        else if (type == typeof(CustomElement))
+        {
+            return new CustomElement
+            {
+                Element = Reference.FromDynamic(source.element),
+                Value = source.value?.ToString(),
+                SearchableValue = source.searchable_value?.ToString()
+            };
+        }
+        else if (type == typeof(SubpagesElement))
         {
-            return new
+            return new SubpagesElement
             {
-                external_id = Element.ExternalId,
+                Element = Reference.FromDynamic(source.element),
+                Value = (source.value as IEnumerable<dynamic>)?.Select<dynamic, Reference>(identifier => Reference.ById(Guid.Parse(identifier.id)))
             };
         }
 
-        throw new DataMisalignedException("Element reference does not contain any identifier.");
+        throw new ArgumentOutOfRangeException($"{type} is not a valid element");
+    }
+
+    private static void ValidateSource(dynamic source)
+    {
+        if (source == null || !DynamicExtensions.HasProperty(source, "element") || !DynamicExtensions.HasProperty(source, "value"))
```

Comment:
Additional check could be `source.GetType().BaseType == typeof(BaseElement)`

Reply:
In my opinion, the basetype doesn't have to be the BaseElement, it is enough to have the required properties